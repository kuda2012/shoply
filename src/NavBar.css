nav {
  width: 100vw;
}
li {
  margin: 3px;
}
