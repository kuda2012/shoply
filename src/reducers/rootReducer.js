const INITIAL_STATE = { cart: {} };

function rootReducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case "ADD_CART":
      let holder = [];
      holder.push(action.payload);
      if (state.cart[action.payload.id]) {
        holder.push(...state.cart[action.payload.id]);
      }
      // console.log({ ...state.cart });
      // console.log({
      //   ...state.cart,
      //   cart: {
      //     [action.payload.id]: holder,
      //   },
      // });
      // return {
      //   ...state,
      //   cart: {
      //     [action.payload.id]: holder,
      //   },
      // };
      let answer;
      if (Object.keys(state.cart).length > 0) {
        for (let key in state.cart) {
          if (key != action.payload.id) {
            answer = {
              cart: {
                ...state.card,
                [action.payload.id]: holder,
              },
            };
            console.log({
              [action.payload.id]: holder,
              ...state.card,
            });
            break;
          } else {
            console.log("dont go here");
            answer = {
              cart: {
                [action.payload.id]: holder,
              },
            };
          }
        }
      } else {
        answer = {
          cart: {
            [action.payload.id]: holder,
          },
        };
      }

      console.log(answer);
      return answer;
    case "REMOVE_CART":
      let gotOne = false;
      let newCart = [];
      for (let i = 0; i < state.cart.length; i++) {
        if (state.cart[i].id == action.payload && !gotOne) {
          gotOne = true;
          newCart = [...state.cart.slice(0, i), ...state.cart.slice(i + 1)];
          break;
        }
      }
      newCart = !gotOne ? [...state.cart] : newCart;

      return {
        ...state,
        cart: newCart,
      };
    default:
      return state;
  }
}

export default rootReducer;
