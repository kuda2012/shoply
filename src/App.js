import React from "react";
import "./App.css";
import Routes from "./Routes";
import NavBar from "./NavBar";

function App() {
  return (
    <div className="App">
      <NavBar />
      <Routes />
    </div>
  );
}

export default App;
