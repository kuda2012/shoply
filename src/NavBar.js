import React from "react";
import "./NavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import { useSelector } from "react-redux";

const NavBar = () => {
  // const { cart } = useSelector((state) => state);
  const cart = [];
  return (
    <div>
      <Navbar expand="md" color="warning">
        <NavLink to="/" className="navbar-brand">
          Shoply
        </NavLink>
        <Nav className="ml-auto">
          <NavItem>
            <NavLink to="/cart">
              Cart {cart.length > 0 ? `(${cart.length})` : ""}
            </NavLink>
          </NavItem>
        </Nav>
      </Navbar>
    </div>
  );
};

export default NavBar;
