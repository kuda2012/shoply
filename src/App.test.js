import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { BrowserRouter } from "react-router-dom";
import rootReducer from "./reducers/rootReducer";
import NavBar from "./NavBar";
const store = createStore(rootReducer);

test("Renders app", () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
});
test("Renders Navbar", () => {
  const { getByText } = render(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>
  );
  expect(getByText("Shoply")).toBeInTheDocument();
});
it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
