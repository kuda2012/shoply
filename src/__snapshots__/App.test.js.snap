// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md bg-warning"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Shoply
      </a>
      <ul
        class="ml-auto nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/cart"
          >
            Cart 
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
